### PR TITLE
Add support for k8s bundles

### PR DESCRIPTION
**Pull request by wallyworld**
Add support for generating the necessary change events for k8s bundles.
For k8s bundles, unit changes and machine changes are omitted. And applications send the num units with the add event, and there's a new scale application change event.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/bundlechanges/144/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/bundlechanges/146/

File: changes_test.go [hunk 1]
```diff
@@ -2171,6 +2323,30 @@ func (s *changesSuite) TestSimpleBundleEmptyModel(c *gc.C) {
...
+                        options:
+                            key-1: value-1
+                            key-2: value-2
+                        annotations:
+                            gui-x: "10"
+                            gui-y: "50"
+            `
+	expectedChanges := []string{
+		"upload charm cs:django-4 for series kubernetes",
+		"deploy application django with 1 unit on kubernetes using cs:django-4",
```
Comment (howbazaar) [inline changes_test.go]: should use two apps, one with scale 2 or more to test output

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/bundlechanges/147/

File: changes.go [hunk 2]
```diff
@@ -59,11 +59,17 @@ func FromData(config ChangesConfig) ([]Change, error) {
 		changes:   changes,
 	}
 	addedApplications := resolver.handleApplications()
-	addedMachines := resolver.handleMachines()
+
+	var addedMachines map[string]*AddMachineChange
+	if resolver.bundle.Type != "kubernetes" {
```
[the remark this answers is not in the document]
Comment (wallyworld) [reply changes.go]: done

File: changes_test.go (hunk 1 above)
Comment (wallyworld) [reply changes_test.go]: done

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/bundlechanges/148/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/bundlechanges/149/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/bundlechanges/150/

**Comment by wallyworld** [comment]
$$merge$$

**Comment by howbazaar** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/bundlechanges-merge

**Pull request merged**